core/src/test/scala/tech/beshu/ror/integration/LdapServerDiscoveryCheckYamlLoadedAccessControlTests.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.integration
+
+import com.dimafeng.testcontainers.ForAllTestContainer
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.Matchers._
+import org.scalatest.{BeforeAndAfterAll, Inside, WordSpec}
+import tech.beshu.ror.accesscontrol.AccessControl.RegularRequestResult
+import tech.beshu.ror.accesscontrol.blocks.Block
+import tech.beshu.ror.accesscontrol.blocks.definitions.ldap.implementations.UnboundidLdapConnectionPoolProvider
+import tech.beshu.ror.accesscontrol.domain.LoggedUser.DirectlyLoggedUser
+import tech.beshu.ror.accesscontrol.domain.User
+import tech.beshu.ror.mocks.MockRequestContext
+import tech.beshu.ror.utils.TestsUtils.{StringOps, basicAuthHeader}
+import tech.beshu.ror.utils.containers.{DnsServerContainer, LdapContainer}
+
+class LdapServerDiscoveryCheckYamlLoadedAccessControlTests
+  extends WordSpec
+    with BaseYamlLoadedAccessControlTest
+    with BeforeAndAfterAll
+    with ForAllTestContainer
+    with Inside {
+
+  private val ldapContainer = new LdapContainer("LDAP1", "test_example.ldif")
+
+  private var dnsContainer: Option[DnsServerContainer] = None
+
+  override val container: LdapContainer = ldapContainer
```

Comment:
maybe we can use MultipleContainers to gather ldap and dnsserver containers together?

Reply:
I wanted to use MultipleContainers, but the problem is that I have to know mapped port of LdapContainer before I start DNS container. MultipleContainers is starting containers in independent way. This is why I'm starting DNS after containers are started

Reply:
OK, but I see we use it in two places so, it'd be nice to have container which behaves like you described. Please take a look at `EsClusterContainer` - this a multi ES containers cluster which uses external dependencies (like LDAP) which are also run in containers but have to be started before starting ES (+ROR obviously) nodes. So this is pretty what you need, but here DnsServerContainer has to wait for LDAP. Maybe we can generalise the EsClusterContainer solution to be ES independent or extract the code you can use in more generic case.